### ISSUE TITLE
[request] pl.describe() improvements

### ISSUE THREAD

**Issue by lcrmorin** [pola-rs/polars#13688]
### Description

The current state of polars describe is quite close to pandas implementation. It has some improvements:
- showing data types
- showing null counts (see https://github.com/pola-rs/polars/issues/4068)

It could be improved further: 
- Counting specific values (typically +/- inf for sklearn interaction, or sentinel values like 999 for interaction with outdated systems). Could work with a user provided list of values to count and maybe +/- inf if they are present, or by default. 
- Count of unique values (by default)
- Constant difference if there is one (typically I work with data of the form 0.1 0.2 ... 0.5, that is rounded floats, I like to know when I can rescale to ints - Maybe by default, or only if there are concerned columns)
- Format options (color) / format the counts as percentage as option provided to users
- Option to add other univariate statistics
- Option to add statistics with respect to another columns (typically correlation with a continuous target, AUC with a binary target ... etc.).

Would those improvements be welcome?

**Comment by JulianCologne** [comment]
I am in favor of a unique count! However, I would call it "distinct count" since unique and distinct are not the same 😀 

For the others I am not sure. I can see them being usefull for specific use cases. Maybe add some parameters to `describe` to optionally add more statistics.

I really like the idea of `describe` being an one-stop-shop for all commonly used statistics instead of sprinkling around 100x cells in the notebook all the time 🤣

**Comment by lcrmorin** [comment]
Yes, I have found describe to not be enough for real life application (you know those hardcoded "999999"). I'd like to get other opinions, then i'll try to handle it.

**Comment by alexander-beedie** [comment]
ref: https://github.com/pola-rs/polars/issues/10270

**Comment by lcrmorin** [comment]
Somehow I just noticed pandas has an info method (https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.info.html) that provides dtypes and non null. counts. Is there something similar in polars ?

**Comment by arv-anshul** [comment]
**Proposal**

Add an additional argument `aggregate_function` to `df.describe()` method where user will pass additional function as `Expr | str | None` to get a better description of DataFrame.

**Example**

```python
df.describe(aggregate_function=[pl.n_unique(), "skew", "kurtosis"])
```

**My Why**: I want to see the "skew" and "kurtosis" of each `numeric` columns at a glance.

**Comment by arv-anshul** [comment]
> Somehow I just noticed pandas has an info method ([pandas.pydata.org/docs/reference/api/pandas.DataFrame.info.html](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.info.html)) that provides dtypes and non null. counts. Is there something similar in polars ?

There is a `df.glimpse()` method which will give you column's datatype and some values of those columns. [See docs.](https://docs.pola.rs/py-polars/html/reference/dataframe/api/polars.DataFrame.glimpse.html)

```python
>>> df.glimpse()
Rows: 3
Columns: 6
$ a  <f64> 1.0, 2.8, 3.0
$ b  <i64> 4, 5, None
$ c <bool> True, False, True
$ d  <str> None, 'b', 'c'
$ e  <str> 'usd', 'eur', None
$ f <date> 2020-01-01, 2021-01-02, 2022-01-01
```
